packages/components/src/context-actions/ContextMenuRoot.tsx

```diff
@@ -56,6 +56,14 @@ class ContextMenuRoot extends Component<
   openMenu: React.RefObject<ContextMenu>;
 
   handleContextMenu(e: MouseEvent): void {
+    let el = e.target as Element | null;
+    while (el != null) {
+      if (el.classList.contains('monaco-editor')) {
```

Comment:
At a high-level, the `@deephaven/components` package shouldn't really have a dependency on `monaco-editor` (or anything monaco really). I see there's already some references to monaco-editor styling in `BaseStyleSheet.scss`, but it doesn't really "belong" there.

As for this... can we instead pass in an array of element classes we should ignore the context menu in? Or pass in a function for whether is should be handled or not? Leaving monaco out of this package entirely.

Reply:
Agreed. Added `ignoredClassNames` prop, moved the class list check to the `ContextActions` component to make this change more decentralized.
I'm exposing the `INPUT_CLASS_NAME` const in `ConsoleInput` instead of just using a wrapper class, as there's an area around the input - paddings, ">" icon on the left, that should still display our custom context menu and not be ignored.